eduNEXT/xblock-filesmanager
# feat allow visibility files by date

## Description
This PR enables users to specify the dates when files/folders will be visible in the LMS. Users can select a range of dates, from a start date to an end date, during which the file/folder will be visible.

**DEMO**


https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/0de9b647-7ca3-40fc-a15d-7b4e6fb27558






### How to test it

- Check this branch`jv/feat-adding-metadata-to-files` in the XBlock files manager

1. use node 18
 ```
nvm use
```

2. install dependencies
```
yarn install 
```
3 run to see any change in the react code 
```
yarn dev 

```
-----
### Setting a date
1. Select a file/folder
2. Right click on "Set file visibility with date" button
![set-vibistily-date](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/07d77cfe-ba98-4412-9f3b-e50a7787ccee)


3. Chose the range of dates
![range-dates](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/d924a853-4f97-45ce-9f0b-976e2e1b0fb4)

4. Click on "Save" button


### Removing a date
1. Select a file/folder
2.  Right click on  "Remove date visibility" button
![remove-dates](https://github.com/eduNEXT/xblock-filesmanager/assets/134975835/7aa37774-a595-4619-9081-ba28272d0ace)

3. Click on "Save" button




**Merge checklist:**
Check off if complete *or* not applicable:
